contracts/Market.sol

```diff
@@ -389,26 +389,40 @@ contract Market is IMarket, Ownable, ERC721 {
 		return _bets[index].payout;
 	}
 
-
 	function _payout(uint256 index, uint8 result) internal virtual {
-		if (result == SCRATCHED) {
-			uint256 lay = _bets[index].payout - _bets[index].amount;
+		uint256 payout = _bets[index].payout;
+		uint256 amount = _bets[index].amount;
+		uint256 loan = payout - amount;
+
+		if (result == SCRATCHED) {				
 			// Transfer the bet amount to the owner of the NFT
-			IERC20(_vault.asset()).transfer(ownerOf(index), _bets[index].amount);
-			// Transfer the lay back to the vault
-			IERC20(_vault.asset()).transfer(address(_vault), lay);
-		} else {
-			address recipient = result == WINNER ? ownerOf(index) : address(_vault);
-			uint256 amount = _bets[index].payout;
+			IERC20(_vault.asset()).transfer(ownerOf(index), amount);
+			// Transfer the loaned amount back to the vault
+			IERC20(_vault.asset()).transfer(address(_vault), loan);
+		}
 
-			// event for vault rewarded
-			if (recipient == address(_vault)) {
-				emit Repaid(recipient, amount);
-			}
+		if (result == WINNER) {
+			// Transfer the payout to the owner of the NFT
+			IERC20(_vault.asset()).transfer(ownerOf(index), payout);
+		}
+			
+		if (result == LOSER) {
+			assert(_vault.getRate() > 0);
+			
+			// Transfer the bet amount plus interest to the vault
+			uint256 repayment = loan * _vault.getRate() / 100_000;
```

Comment:
Is it possible that the amount to repay > loan + bet amount?

Reply:
hmm, if odds are less than the rate say $1.06

Reply:
Tested and discussed.